examples/http/azure_containers.js

```diff
@@ -0,0 +1,552 @@
+ * @param {Array} containerGroupInfoList - A list of Container Group information objects.
+ * @returns {Array} The filtered list of Container Group information based on resource groups.
+ */
+function filterContainerGroupInfoListByResourceGroups(containerGroupInfoList) {
+    if (!(resourceGroups.length === 1 && resourceGroups[0].toLowerCase() === 'all')) {
+        return containerGroupInfoList.filter(function (containerGroup) {
+            return resourceGroups.some(function (group) {
+                return group.toLowerCase() === containerGroup.resourceGroup.toLowerCase()
+            })
+        });
+    }
+    return containerGroupInfoList;
+}
+
+/**
+ * Filters the list of Container Group information by specified container names.
+ * @param {Array} containerGroupInfoList - A list of Container Group information objects.
+ * @returns {Array} The filtered list of Container Group information based on Container names.
+ */
+function filterContainerGroupInfoListByContainerNames(containerGroupInfoList) {
+    if (!(containerNames.length === 1 && containerNames[0].toLowerCase() === 'all')) {
+        return containerGroupInfoList.filter(function (containerGroup) {
+            return containerNames.some(function (containerName) {
+                return containerName.toLowerCase() === containerGroup.containerName.toLowerCase();
+            });
+        });
+    }
+    return containerGroupInfoList;
+}
+
+/**
+ * Filters the list of Container Group information by specified container names.
+ * @param {Array} containerGroupInfoList - A list of Container Group information objects.
+ * @returns {Array} The filtered list of Container Group information based on Container names.
+ */
+function filterContainerGroupHadContainers(containerGroupInfoList) {
+    return containerGroupInfoList.filter(function (containerGroup) {
+        return containerGroup.containers.length;
+    });
+}
+
+/**
+ * Processes the response from the Container Groups API call, extracts Container Groups data, and populates the table.
+ * @param {Object} d - The deferred promise object.
+ * @returns {Function} A function to process the HTTP response.
+ */
+function processContainerGroupsResponse(d) {
+    return function process(error, response, body) {
+        checkHTTPError(error, response);
+        const bodyAsJSON = JSON.parse(body);
+        if (!bodyAsJSON.value) {
+            D.failure(D.errorType.GENERIC_ERROR)
+            d.reject("No Container Groups found in the response");
+            return;
+        }
+        let containerGroupInfoList = bodyAsJSON.value.map(extractContainerGroupInfo);
+        if (!containerGroupInfoList.length) {
+            console.info('There is no Container Group');
+        } else {
+            containerGroupInfoList = filterContainerGroupInfoListByResourceGroups(filterContainerGroupInfoListByContainerNames(filterContainerGroupHadContainers(containerGroupInfoList)));
```

Comment:
why we should iterate over and over and over 3 times, we can iterate with only one filter and make our 3 conditions